### DISCUSSION TITLE
Preview generation is really slow

### DISCUSSION THREAD

**Discussion by jacksbox** [nextcloud/all-in-one#3111] [Questions] [answered]
I'm trying to generate previews for 60k images. The process is unbelievably slow.
Is there anything I can do to speed up this process?

### Steps to reproduce
1. Run the generate previews occ command

### Expected behavior

Preview images get generate in a reasonable time

### Actual behavior 

The Imaginary container creates roughly 12 Images per Minute... for the 5+ Previews for every of the 60k images, that will take way more than 10 days to complete.

I do not expect something like 10min for all 60k pictures... but also not 10 days.

### Host OS 

Ubuntu 22.04

#### Nextcloud AIO version 

6.4.0

#### Current channel 

latest

#### Other valuable info 

I'm using not a very beefy system, 1vCPU and 2GB RAM, but here is a snapshot of `docker stats`, the CPU usage varies, but most of the below 100% total. Memory usage is way below 100%.

I also tested to increase the core count to 2, but the preview generation speed does not improve at all.

That leads me to the assumption that my system is not the bottleneck - neither CPU nor RAM seem to be used fully...

```
CONTAINER ID   NAME                            CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O         PIDS
0ae106148f4c   nextcloud-aio-apache            0.04%     54.75MiB / 1.883GiB   2.84%     5.87MB / 34.4MB   219MB / 0B        120
50c99b6cf403   nextcloud-aio-notify-push       0.06%     3.688MiB / 1.883GiB   0.19%     3.08MB / 1.03MB   25.8MB / 0B       3
f43e5b643587   nextcloud-aio-nextcloud         7.37%     338.7MiB / 1.883GiB   17.57%    978MB / 12.4GB    809MB / 162MB     7
2b6ad29c0177   nextcloud-aio-imaginary         0.00%     111MiB / 1.883GiB     5.76%     12.2GB / 705MB    42.5MB / 0B       12
17628395b546   nextcloud-aio-redis             0.23%     4.816MiB / 1.883GiB   0.25%     39.5MB / 17.9MB   16MB / 1.57MB     5
6c8bfcd93c97   nextcloud-aio-database          1.13%     226.1MiB / 1.883GiB   11.73%    204MB / 203MB     2.94GB / 1.41GB   8
b39c42e675d4   nextcloud-aio-mastercontainer   0.52%     50.7MiB / 1.883GiB    2.63%     3.77MB / 15.3MB   216MB / 508kB     144
```

**Comment by szaimen** [comment]
Hi, I suspect the cpu to be the bottlneck here. Can you upgrade to e.g. 4-cores and check if it improves the speed?

**Reply by jacksbox** [reply]
Nope, sadly that does not help. I upgrade to 2 cores, speed is exactly the same 12 images per minute / 5 sec per image.

**Comment by szaimen** [comment, marked as answer]
Hi, can you upgrade to v7.0.0 Beta and check if that resolves things? See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by meichthys** [reply]
I'm on the latest tag for imaginary and I'm also experiencing much slower preview generation using imaginary vs the older PreviewGenerator. It's not much of an issue for me, and I don't think it will have a hard time keeping up once it finishes the initial generation, but the initial generation is taking much longer than the original PreviewGenerator which took a few hours.
